lib/puppet_blacksmith/modulefile.rb

```diff
@@ -1,49 +1,25 @@
-# Avoid loading everything Puppet when we only need the module tool
-# It causes trouble running it from jruby warbled jars
-# module Puppet
-#   def self.settings
-#     {}
-#   end
-#   def self.[](param)
-#   end
-# end
-# require 'puppet/error'
-# require 'puppet/module_tool'
-require 'puppet'
-require 'puppet/module_tool'
 require 'puppet_blacksmith/version_helper'
 
-Puppet[:confdir] = "."
-
 module Blacksmith
   class Modulefile
 
-    FILES = ["metadata.json", "Modulefile"]
+    FILES = ["metadata.json"]
 
     attr_reader :path
 
     def initialize(path = nil)
       @path = path.nil? ? FILES.find {|f| File.exists? f} : path
       raise Blacksmith::Error, "Unable to find any of #{FILES}" unless @path
-      @modulefile = @path =~ /Modulefile$/
     end
 
     def metadata
-      unless @metadata
-        if @modulefile
-          metadata = Puppet::ModuleTool::Metadata.new
-          Puppet::ModuleTool::ModulefileReader.evaluate(metadata, path)
-          @metadata = { 'name' => metadata.name, 'version' => metadata.version, 'author' => metadata.author }
-        else
-          @metadata = JSON.parse(File.read(path))
-        end
-      end
+      @metadata = JSON.parse(File.read(path)) unless @metadata
```

Comment:
couldn't you remove the instance variable entirely here and just have the method return the parsed JSON?

Reply:
Currently it provides caching of the metadata and this is called in quite a few places. If there is a better way to provide the same feature I'd love to hear it.